docs/api/v4/usage.md

```diff
@@ -0,0 +1,196 @@
+
+```php
+[
+ { /* DAO Object */
+]
+```
+
+The result of a failed API call typically looks like this:
+
+```php
+array(
+  'is_error' => 1,
+  'error_message' => /* a descriptive error message */,
+)
+```
+
+!!! note
+    A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
+
+
+## PHP
+
+This is the most common way to call the API. There are 2 formats of API calls a proceedural approach and the more traditional approach
+
+Object Oriented Proceedural approach:
+
+```php
+$contacts = \Civi\Api4\Contact::get()
+  ->addWhere('last_name', '=', 'Adams')
+  ->setLimit(25)
+  ->execute();
+```
+
+Note that we use the a format of Object::action and then there are helper functions like AddWhere
+
+```php
+try {
+  $contacts = civicrm_api4('Contact', 'get', array(
+    'where' => [
+      ['first_name', '=', 'Alice'],
+      ['last_name', '=', 'Roberts'],
+    ],
+  ]);
+}
+catch (\API_Exception $e) {
+  $error = $e->getMessage();
+}
+printf("Found %d item(s)\n", count($contacts));
+```
+
+This format matches canonical format almost exactly, with a few improvements for usability:
+
+-  The `version => 4` parameter is not required.
+-  Errors are reported as PHP exceptions. You may catch the exceptions or (by default) allow them to bubble up.
+- You can immediately iterate over the contacts returned
+
+*Note*: If you're writing a Drupal module, a Joomla extension, a WordPress plugin, or a standalone script, then you may need to **bootstrap** CiviCRM before using the API.  See the examples in [Bootstrap Reference].
+
+[Bootstrap Reference](/framework/bootstrap.md)
+
+## REST
+
+For external services:
+
+```text
+http://www.example.com/sites/all/modules/civicrm/extern/rest.php
+  ?api_key=t0ps3cr3t
+  &key=an07h3rs3cr3t
+  &json=1
+  &debug=1
+  &version=4
+  &entity=Contact
+  &action=get
+  &first_name=Alice
+  &last_name=Roberts
+```
+
+For sessions already authenticated by the CMS (e.g. AJAX)
+
+```text
+http://www.example.com/civicrm/ajax/rest
+  ?json=1
+  &debug=1
+  &version=3
+  &entity=Contact
+  &action=get
+  &first_name=Alice
+  &last_name=Roberts
+```
+
+Obviously you should substitute your site in! You can explore the syntax and options available using the [API Explorer](/api/index.md#api-explorer).
+
+Please note that the REST interface is subject to [API Security](/security/permissions.md#api-permissions).
+
+For more details, see [REST interface](/api/interfaces.md#rest). 
+
+
+## AJAX
+
+```javascript
+CRM.api4('entity', 'action', [params], [statusMessage]);
+```
+
+If you pass `true` in as the `StatusMessage` param, it will display the default status Message. This is useful when doing things such as adding tags to contacts or similar. If you wish to do further work based on the result of the API call (e.g use the results from a GET call) you will need to use the [done method](http://api.jquery.com/deferred.done/) to listen for the event. For example:
```

Comment:
```suggestion
If you pass `true` in as the `StatusMessage` param, it will display the default Status Message. This is useful when doing things such as adding tags to contacts or similar. If you wish to do further work based on the result of the API call (e.g use the results from a GET call) you will need to use the [done method](http://api.jquery.com/deferred.done/) to listen for the event. For example:
```